File: agent_misc/mcps/codeql-development-mcp-server/server/ql/cpp/tools/test/CallGraphTo/Example1.cpp
// COMPLIANT: Functions that should not be analyzed
void unrelated1() {
    // No calls
}

// NON_COMPLIANT: Target function for call graph analysis
void targetFunc() {
    unrelated1();
}

void caller1() {
    targetFunc();
}

void caller2() {
    targetFunc();
}

File: agent_misc/mcps/codeql-development-mcp-server/server/ql/rust/tools/test/CallGraphTo/Example1.rs
// COMPLIANT: Functions that should not be analyzed
fn unrelated1() {
    // No calls
}

// NON_COMPLIANT: Target function for call graph analysis
fn target_func() {
    unrelated1();
}

fn caller1() {
    target_func();
}

fn caller2() {
    target_func();
}

fn main() {
    caller1();
    caller2();
}
